### DISCUSSION TITLE
in output file，I get Px and weight，the particle is electron,can I calculate the velocity by  （px/weight/9.01e-31）?

### DISCUSSION THREAD

**Discussion by XnjeMa** [epochpic/epoch#712] [Q&A]
in output file，I get Px and weight，the particle is electron,can I calculate the velocity by  （px/weight/9.01e-31）?

**Comment by Status-Mirror** [comment]
Hi @XnjeMa,

You're right that EPOCH uses macro-particles which have weights, but when EPOCH outputs particle information like $p_x$, this refers to a single real particle - you don't need to divide by weight.

EPOCH will quote the relativistic momentum though, so instead of $p_x = m v_x$, it's actually $p_x = \gamma m v_x$.

To get the speed from a relativistic momentum, you can use:

- $p^2 = p_x^2 + p_y^2 + p_z^2$
- $E = \sqrt{p^2c^2 + m^2c^4}$
- $v/c = pc/E$

Where the first equation calculates the magnitude of momentum, the second is the relativistic energy equation, and the third comes from the definitions of $E=\gamma m c^2$ and $p = \gamma m v$.

Hope this helps,
Stuart
